samples/timer_watchdog/src/main.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/console/console.h>
+#include <zephyr/drivers/gpio.h>
+#include <npmx_core.h>
+#include <npmx_driver.h>
+#include <npmx_gpio.h>
+#include <npmx_timer.h>
+
+#define LOG_MODULE_NAME watchdog
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/** @brief Variable filled in @ref kicking_thread() when watchdog kicking stops. */
+static volatile uint32_t start_time;
+
+/** @brief Variable filled in @ref timer_callback() when watchdog warning occurs. */
+static volatile uint32_t warning_time;
+
+/** @brief Semaphore used to inform @ref kicking_thread() when to stop kicking the watchdog. */
+K_SEM_DEFINE(stop_kick_sem, 0, 1);
+
+/** @brief Stack used by kick_thread. */
+K_THREAD_STACK_DEFINE(kick_stack, 1000);
+
+/** @brief Kicking thread definition. */
+struct k_thread kick_thread;
+
+#if CONFIG_TESTCASE_WATCHDOG_MODE_WARNING
+/** @brief Timer watchdog working in warning mode. */
+#define TIMER_WATCHDOG_MODE NPMX_TIMER_MODE_WATCHDOG_WARNING
+#elif CONFIG_TESTCASE_WATCHDOG_MODE_RESET
+/** @brief Timer watchdog working in reset mode. */
+#define TIMER_WATCHDOG_MODE NPMX_TIMER_MODE_WATCHDOG_RESET
+#endif
```

Comment:
How can those options be set? Kconfig is empty.

Reply:
Kconfig is not empty:
```
mainmenu "Timer watchdog sample."

choice
	bool "Watchdog mode selection."

config TESTCASE_WATCHDOG_MODE_WARNING
	bool "Set watchdog to warning mode."

config TESTCASE_WATCHDOG_MODE_RESET
	bool "Set watchdog to reset mode."

endchoice

source "Kconfig.zephyr"
```